rest/admin_api_test.go

```diff
@@ -33,6 +33,121 @@ import (
+			channelLengthToAdd: 0,
+			expectWarn:         false,
+		},
+		{
+			name:               "Under max channel length",
+			channelLengthToAdd: -1,
+			expectWarn:         false,
+		},
+	}
+
+	boundaryTest := func(warnThresholdLength int) {
+		for _, test := range boundaryTestCases {
+			channelLength := warnThresholdLength + test.channelLengthToAdd
+			t.Run(fmt.Sprintf("%s: %d", test.name, channelLength), func(t *testing.T) {
+				chanNameWarnCountBefore := getChannelNameWarnCount()
+				tr := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/doc%v", channelLength), fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("A", channelLength)))
+				assertStatus(t, tr, http.StatusCreated)
+				chanNameWarnCountAfter := getChannelNameWarnCount()
+				if test.expectWarn {
+					assert.Equal(t, chanNameWarnCountBefore+1, chanNameWarnCountAfter)
+				} else {
+					assert.Equal(t, chanNameWarnCountBefore, chanNameWarnCountAfter)
+				}
+			})
+		}
+	}
+
+	boundaryTest(int(base.DefaultWarnThresholdChannelNameSize))
+
+	// Update doc - should warn
+	channelLength := int(base.DefaultWarnThresholdChannelNameSize) + 5
+	t.Run("Update doc without changing channel", func(t *testing.T) {
+		tr := rt.SendAdminRequest("PUT", "/db/replace", fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("B", channelLength))) // init doc
+		assertStatus(t, tr, http.StatusCreated)
+
+		before := getChannelNameWarnCount()
+		tr = rt.SendAdminRequest("PUT", "/db/replace?rev="+getRespRev(t, tr), fmt.Sprintf("{\"chan\":\"%s\", \"data\":\"test\"}", strings.Repeat("B", channelLength)))
+		assertStatus(t, tr, http.StatusCreated)
+		after := getChannelNameWarnCount()
+		assert.Equal(t, before+1, after)
+	})
+
+	// Update doc channel with creation of a new channel
+	t.Run("Update doc with new channel", func(t *testing.T) {
+		tr := rt.SendAdminRequest("PUT", "/db/replaceNewChannel", fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("C", channelLength))) // init doc
+		assertStatus(t, tr, http.StatusCreated)
+
+		before := getChannelNameWarnCount()
+		tr = rt.SendAdminRequest("PUT", "/db/replaceNewChannel?rev="+getRespRev(t, tr), fmt.Sprintf("{\"chan\":\"%s\", \"data\":\"test\"}", strings.Repeat("D", channelLength+5)))
+		assertStatus(t, tr, http.StatusCreated)
+		after := getChannelNameWarnCount()
+		assert.Equal(t, before+1, after)
+	})
+
+	// Delete channel over max len - no warning
+	t.Run("Delete channel over max length", func(t *testing.T) {
+		tr := rt.SendAdminRequest("PUT", "/db/deleteme", fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("F", channelLength))) // init channel
+		assertStatus(t, tr, http.StatusCreated)
+
+		before := getChannelNameWarnCount()
+		tr = rt.SendAdminRequest("DELETE", "/db/deleteme?rev="+getRespRev(t, tr), "")
+		assertStatus(t, tr, http.StatusOK)
+		after := getChannelNameWarnCount()
+		assert.Equal(t, before, after)
+	})
+
+	// change value to 500 in config
+	warnThreshold := uint32(500)
+	rt = NewRestTester(t, &RestTesterConfig{
```

Comment:
I think this is going to create a new rest tester with the new config rather than just modify the config like you want. 
I think you should be able to just set it with `rt.DatabaseConfig.Unsupported.WarningThresholds`....

Reply:
Had problems getting this modification to work as for some reason it wasn't recognising the change through the "db.Options.UnsupportedOptions.WarningThresholds.ChannelNameSize" line in crud.go.

I have made it so a new rest tester is made per loop iteration which might be quite intensive to do, but I think it will be a better way to test it as, I assume, the user will be setting the Warning Threshold in the config before SG starts. 

Let me know if that seems like an ok way to do it or if there's something I am missing